toktagger/api/crud/utils.py

```
@@ -252,3 +253,37 @@ async def get_sample_summary(
         summary.data.file_name = str(Path(summary.data.file_name).parent)
 
     return summary
+
+
+async def import_annotations(
+    db_client: MongoDBClient,
+    project_id: str,
+    annotations: list[AnnotationTypes],
+) -> None:
+    ids = {
+        "project_id": convert_to_objectid(project_id, "projects"),
+    }
+
+    if not await db_client.get_document_by_id("projects", ids["project_id"]):
+        raise HTTPException(status_code=404, detail="Project not found with that ID.")
+
+    if len(annotations) == 0:
+        return
+
+    sample_groups = defaultdict(list)
+    for annotation in annotations:
+        sample_groups[annotation.sample_id].append(annotation)
+
+    for sample_id, sample_annotations in sample_groups.items():
+        sample_obj_id = convert_to_objectid(sample_id, "samples")
+
+        if not await db_client.get_document_by_id("samples", sample_obj_id):
```

Do we want this to fail on the first invalid sample ID? Or continue trying all others, then throw one error at the end with a list of any invalid sample IDs?

---

I think fail here. Something has gone wrong at this point.